Dev

DenisFilichev opened the pull request:
Добавлено добавление и удаление товаров и заказов

Commit:
Добавлена работа с SQL:
1. Чтение товаров и заказов из SQL
2. Запись новых товаров в SQL
3. Запись новых заказов в SQL (без привязки к товарам)

Commit:
Добавлена функции с SQL:
1. Добавление и удаление товаров в заказы с корректировкой количества на складе
2. Удаление заказов

konivex reviewed (changes requested):
Хранилище данных — файлы или БД — должно определяться настройками файла свойств, а не исходя из попытки подключения к БД. Если файл свойств настроен на БД, и к БД не подключиться, то должно выдаваться сообщение об ошибке, и приложение должно завершать работу.

Затем, FileListOrders и DBOrders д.б. объединены в иерархию с общим абстрактным классом, тогда при запуске в зависимости от настроек вы создаёте экземпляр того или другого класса, после чего остальная часть приложения даже и не будет знать (да это и не нужно), какой тип хранилища используется.

Использование шаблона операторов в виде "insert into ... values ('%s'" не безопасно! Нужно использовать PreparedStatement.

Commit:
Исправлены замечания:
1. Создан файл properties, с установкой места хранилища
2. Создан абстрактный класс для записи и чтения файлов и базы

Commit:
Исправлены замечания:
1. Заменены операции Statement на PreparedStatement
2. Исправлена ошибка сохранения товаров в SQL и в файл.

konivex reviewed (changes requested):
Не нашёл sql-файла со скриптом создания таблиц БД (две-три тестовые записи в каждую таблицу тоже было бы неплохо).

Commit:
Добавлена возможность создания дефолтных таблиц, но без проверки существующих таблиц

DenisFilichev committed:
Скрипт для создания таблиц и пары примеров наполнения

konivex reviewed (changes requested):
Почему при нажатии кнопки "добавить товар в заказ" список товаров в открывшемся окне пуст, хотя судя по таблицам БД какие-то остатки на складе ещё есть?
Исправьте, чтобы остатки корректно выводились в окно добавления позиции в заказ, чтобы можно было добавлять товар в заказ.

DenisFilichev commented:
> Почему при нажатии кнопки "добавить товар в заказ" список товаров в открывшемся окне пуст, хотя судя по таблицам БД какие-то остатки на складе ещё есть?
> Исправьте, чтобы остатки корректно выводились в окно добавления позиции в заказ, чтобы можно было добавлять товар в заказ.

Добавление товара в заказ работает.
Возможно вы добавляете уже существующий в заказе товар, в этом случае меняется его количество и на складе соответственно количество уменьшается.

konivex reviewed (approved):
Все поля д.б. private!
Других замечаний нет. Зачтено.
